service/history/task/redispatcher.go

```diff
@@ -261,44 +253,21 @@ func (r *redispatcherImpl) redispatchTasks(notification redispatchNotification)
 				totalRedispatched++
 			}
 		}
+		if !pq.IsEmpty() {
+			item, _ := pq.Peek()
+			r.timerGate.Update(item.redispatchTime)
```

Comment:
Check if `Peek()` returned error. `item` might be nil

Reply:
Peek never returns error if it's not empty.
Maybe I can introduce a new method `MustPeek`, which is only used on non-empty queue